mm/2s2h/Enhancements/Sfx/MuteCarpenterSfx.cpp

```diff
@@ -0,0 +1,28 @@
+#include <libultraship/bridge.h>
+#include "2s2h/GameInteractor/GameInteractor.h"
+#include "2s2h/ShipInit.hpp"
+
+extern "C" {
+#include "overlays/actors/ovl_Obj_Sound/z_obj_sound.h"
+}
+
+#define CVAR_NAME "gEnhancements.Sfx.MuteCarpenterSfx"
+#define CVAR CVarGetInteger(CVAR_NAME, 0)
+
+void RegisterMuteCarpenterSfx() {
+    COND_ID_HOOK(ShouldActorInit, ACTOR_OBJ_SOUND, CVAR, [](Actor* actor, bool* should) {
+        ObjSound* objSound = (ObjSound*)actor;
+
+        // Returning if not in South or East Clocktown just as a safety precaution, incase these actor params are used
+        // elsewhere
+        if (gPlayState->sceneId != SCENE_CLOCKTOWER && gPlayState->sceneId != SCENE_TOWN) {
+            return;
+        }
+
+        if ((objSound->actor.params & OBJ_SOUND_ID_MASK) == 16) {
+            Actor_Kill(&objSound->actor);
```

Comment:
[nit]: minor comment about the behavior of the `ShouldActorInit` hook, setting `*should = false` will prevent the actor from spawning (by calling `Actor_Kill()` for you), but it also ensures that the actual actors init func is never executed.

In this case, nothing significant is happening in obj_sound init func, but it would be good practice to set the should result instead.

```suggestion
            *should = false;
```